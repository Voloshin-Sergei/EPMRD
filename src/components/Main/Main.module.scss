@import 'Styles/variables/colors';
@import 'Styles/variables/fonts';

.main {
  flex: 1 1 auto;
}

.error {
  color: $color-secondary;
  font-weight: $font-weight--medium;
  font-size: 6rem;
  text-align: center;
}
