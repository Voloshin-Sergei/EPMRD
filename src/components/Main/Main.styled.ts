import styled from 'styled-components';
import { colors } from '../../styles/variables/colors.styled';
import { fonts } from '../../styles/variables/fonts.styled';

export const StyledMain = styled.main`
  flex: 1 1 auto;
`;
export const StyledError = styled.h2`
  color: ${colors.colorSecondary};
  font-weight: ${fonts.fontWeightMedium};
  font-size: 6rem;
  text-align: center;
`;

export const StyledTitle = styled.h2`
  font-weight: ${fonts.fontWeightMedium};
  font-size: 6rem;
  text-align: center;
`;
