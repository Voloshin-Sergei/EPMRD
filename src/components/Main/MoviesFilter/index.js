export { MoviesFilter as default } from './MoviesFilter';
