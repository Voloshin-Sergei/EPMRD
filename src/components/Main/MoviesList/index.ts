export { MoviesList } from './MoviesList';
export { Movie } from './MoviesList';