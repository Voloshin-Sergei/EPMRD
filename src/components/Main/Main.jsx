import React from 'react';
import Sort from './Sort/Sort.jsx';
import MoviesList from './MoviesList/MoviesList.jsx';

const Main = () => (
  <>
    <Sort />
    <MoviesList />
  </>
);

export default Main;
