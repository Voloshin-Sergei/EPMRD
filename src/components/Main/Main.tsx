import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMovies } from 'Store/actions/searchMovieAction';
import { RootState } from 'Store/reducers';
import { Loader } from 'Components/common/Loader';
import { MoviesList } from './MoviesList';
import { StyledMain, StyledError, StyledTitle } from './Main.styled';

export interface Movie {
  id: number;
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
}

export const Main: React.FC = () => {
  const { isLoading, error, movies, filter, category, inputValue } = useSelector(
    (state: RootState) => state.searchMovieReducer,
  );

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMovies(filter, category, inputValue));
  }, [fetchMovies, filter, category, inputValue]);

  const renderMovieList = () => {
    if (error && error instanceof Error) {
      return <StyledError>{error.message}</StyledError>;
    }
    if (isLoading) {
      return <Loader />;
    }

    if (!movies.length) {
      return <StyledTitle>No films found</StyledTitle>;
    }
    return <MoviesList movieList={movies} />;
  };

  return <StyledMain>{renderMovieList()}</StyledMain>;
};
