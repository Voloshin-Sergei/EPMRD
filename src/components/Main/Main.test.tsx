import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Main }  from './Main';
Enzyme.configure({ adapter: new Adapter() });

describe("Main component", () => {
    it('should render Main component', () => {
    const component = render(<Main/>);  
    expect(component).toMatchSnapshot();
  });
});