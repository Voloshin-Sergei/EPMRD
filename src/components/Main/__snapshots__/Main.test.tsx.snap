// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render Main component 1`] = `
Array [
  <div
    class="filter"
  >
    <div>
      <p
        class="result"
      >
        7 movies found
      </p>
    </div>
    <div
      class="buttons"
    >
      <p
        class="text"
      >
        Sort by
      </p>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <button
            class="button tag secondary"
            type="button"
          >
            release
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="button tag secondary"
            type="button"
          >
            date
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="button tag secondary"
            type="button"
          >
            rating
          </button>
        </li>
      </ul>
    </div>
  </div>,
  <section
    class="movies"
  >
    <div
      class="movie"
    >
      <img
        alt="Black Widow"
        class="poster"
        src="https://image.tmdb.org/t/p/w500/qAZ0pzat24kLdO3o8ejmbLxyOac.jpg"
      />
      <div
        class="info"
      >
        <h3
          class="title"
        >
          Black Widow
        </h3>
        <span
          class="year"
        >
          2021
        </span>
      </div>
      <ul
        class="genres"
      >
        <li
          class="genre"
        >
          Action
        </li>
        <li
          class="genre"
        >
          Adventure
        </li>
        <li
          class="genre"
        >
          Thriller
        </li>
        <li
          class="genre"
        >
          Science Fiction
        </li>
      </ul>
    </div>
    <div
      class="movie"
    >
      <img
        alt="The Boss Baby: Family Business"
        class="poster"
        src="https://image.tmdb.org/t/p/w500/5dExO5G2iaaTxYnLIFKLWofDzyI.jpg"
      />
      <div
        class="info"
      >
        <h3
          class="title"
        >
          The Boss Baby: Family Business
        </h3>
        <span
          class="year"
        >
          2021
        </span>
      </div>
      <ul
        class="genres"
      >
        <li
          class="genre"
        >
          Animation
        </li>
        <li
          class="genre"
        >
          Comedy
        </li>
        <li
          class="genre"
        >
          Adventure
        </li>
        <li
          class="genre"
        >
          Family
        </li>
      </ul>
    </div>
    <div
      class="movie"
    >
      <img
        alt="The Tomorrow War"
        class="poster"
        src="https://image.tmdb.org/t/p/w500/34nDCQZwaEvsy4CFO5hkGRFDCVU.jpg"
      />
      <div
        class="info"
      >
        <h3
          class="title"
        >
          The Tomorrow War
        </h3>
        <span
          class="year"
        >
          2021
        </span>
      </div>
      <ul
        class="genres"
      >
        <li
          class="genre"
        >
          Action
        </li>
        <li
          class="genre"
        >
          Science Fiction
        </li>
      </ul>
    </div>
    <div
      class="movie"
    >
      <img
        alt="Space Jam: A New Legacy"
        class="poster"
        src="https://image.tmdb.org/t/p/w500/5bFK5d3mVTAvBCXi5NPWH0tYjKl.jpg"
      />
      <div
        class="info"
      >
        <h3
          class="title"
        >
          Space Jam: A New Legacy
        </h3>
        <span
          class="year"
        >
          2021
        </span>
      </div>
      <ul
        class="genres"
      >
        <li
          class="genre"
        >
          Animation
        </li>
        <li
          class="genre"
        >
          Comedy
        </li>
        <li
          class="genre"
        >
          Family
        </li>
        <li
          class="genre"
        >
          Science Fiction
        </li>
      </ul>
    </div>
    <div
      class="movie"
    >
      <img
        alt="Luca"
        class="poster"
        src="https://image.tmdb.org/t/p/w500/jTswp6KyDYKtvC52GbHagrZbGvD.jpg"
      />
      <div
        class="info"
      >
        <h3
          class="title"
        >
          Luca
        </h3>
        <span
          class="year"
        >
          2021
        </span>
      </div>
      <ul
        class="genres"
      >
        <li
          class="genre"
        >
          Animation
        </li>
        <li
          class="genre"
        >
          Comedy
        </li>
        <li
          class="genre"
        >
          Family
        </li>
        <li
          class="genre"
        >
          Fantasy
        </li>
      </ul>
    </div>
    <div
      class="movie"
    >
      <img
        alt="F9"
        class="poster"
        src="https://image.tmdb.org/t/p/w500/bOFaAXmWWXC3Rbv4u4uM9ZSzRXP.jpg"
      />
      <div
        class="info"
      >
        <h3
          class="title"
        >
          F9
        </h3>
        <span
          class="year"
        >
          2021
        </span>
      </div>
      <ul
        class="genres"
      >
        <li
          class="genre"
        >
          Action
        </li>
        <li
          class="genre"
        >
          Adventure
        </li>
        <li
          class="genre"
        >
          Crime
        </li>
        <li
          class="genre"
        >
          Thriller
        </li>
      </ul>
    </div>
    <div
      class="movie"
    >
      <img
        alt="The Forever Purge"
        class="poster"
        src="https://image.tmdb.org/t/p/w500/uHA5COgDzcxjpYSHHulrKVl6ByL.jpg"
      />
      <div
        class="info"
      >
        <h3
          class="title"
        >
          The Forever Purge
        </h3>
        <span
          class="year"
        >
          2021
        </span>
      </div>
      <ul
        class="genres"
      >
        <li
          class="genre"
        >
          Adventure
        </li>
        <li
          class="genre"
        >
          Thriller
        </li>
        <li
          class="genre"
        >
          Action
        </li>
        <li
          class="genre"
        >
          Horror
        </li>
        <li
          class="genre"
        >
          Western
        </li>
      </ul>
    </div>
  </section>,
]
`;
